src/config.rs

```diff
@@ -293,6 +293,20 @@ pub trait Clock {
     /// it represents a sequence that can attribute a logical ordering
     /// to actions on the database.
     fn now(&self) -> i64;
+
+    /// SlateDB enforces monotonic clocks, even across restarts, which can result in situations where
+    /// the persisted time (from the manifest or WAL) is ahead of the current time on a machine.
+    /// In such cases, this method calculates how long the system should sleep to allow the clock
+    /// to synchronize with the last persisted time.
+    ///
+    /// # Parameters
+    ///
+    /// - `last_tick`: The last observed persisted clock tick (in milliseconds).
```

Comment:
I'm a little confused about this. Are clock units always going to be time based? I think we were talking about using logical clocks like Kafka offsets as well

Reply:
Yup you're right there, the docs should be updated. This is exactly why (by the way) I wanted a configurable "wait" duration. For logical clocks I suspect that most implementations of `clock_sync_duration` should just return a Duration of zero (as they should always just be correctly configured and a thread sleep generally won't help)

Reply:
Yea. In the interest of simplicity, I feel like dropping the configurable duration is still the right call.  Like.. can we just sleep for 2 seconds and try again, then barf if it still doesn't work?

Reply:
Yeah, I'm in agreement. I think to make it a little nicer I'll sleep for 
```rs
cmp::min(10_000, 2 * (last_tick - tick).abs() as u64)
```
Basically sleep no more than 10 seconds, but if the difference is less than that just wait 2x the difference (so if it's 1 second difference it'll just sleep for 2)